### ISSUE TITLE
Allow Submenus in Navigation Overlay to respect "Toggle on Click" 

### ISSUE THREAD

**Issue by Greatdane** [WordPress/gutenberg#44346]
### Description

In a navigation block with a submenu, when selecting Open on click and Always Overlay, the submenu items do not collapse and the open to show an arrow is not selectable. The arrow is also not present.

This behavior continues on the live page. It looks like there is a visibility rule that doesn't get toggled in this view. Here's a workaround with CSS:

```
.wp-block-navigation-item.open-on-click .wp-block-navigation-submenu__toggle[aria-expanded="false"] ~ ul.wp-block-navigation__submenu-container {
    visibility: hidden;
}
```

### Step-by-step reproduction instructions

1) Create navigation block with submenu
2) Select overlay Always and toggle On-click
3) Save, submenus always active, no arrow visible.

### Screenshots, screen recording, code snippet

<img width="1162" alt="Screenshot on 2022-09-18 at 14-54-41" src="https://user-images.githubusercontent.com/71042895/190923870-842d74c3-5ac3-450b-92dd-130f85dd1e6d.png">

### Environment info

WordPress 6.01
Gutenberg 14.1.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
Hi!
This is the intended behavior of the navigation block. I mean it is intentional that submenus are never collapsed in the overlay, regardless of the block settings.

Perhaps the issue could be rephrased slightly, and be a _feature request_ for the behavior you want the submenus to have?

**Comment by antonyjsmith** [comment]
The current behaviour makes little sense to me, submenus should be collapsed on mobile the same as Desktop. The setting are confusing since they do not indicate that this setting applies to desktop only. Especially since when using "always" for overlay the submenu "open on click/show arrows" option is still there despite it doing nothing,

**Comment by masteradhoc** [comment]
+1 on this. currently there's very little love put into the mobile menu...

**Comment by gaswirth** [comment]
+1 as well. The navigation block is still, as far as I'm concerned, just not ready. Forcing submenus to be always open does not make sense, but even worse, the settings interface doesn't indicate that some options simply don't work with certain combinations, like setting "Overlay" to "Always."

**Comment by kathrynwp** [comment]
Another report from a user who wishes sub-menus would be collapsed by default on mobile:

https://wordpress.org/support/topic/collapse-sub-menu-on-mobile/

**Comment by mgozdis** [comment]
+1 Feature requested by user.

**Comment by phil-sola** [comment]
+1 for this. The mobile navigation out of the box generally needs a lot of work but to leave the sub menu open by default makes no sense to me whatsoever and not something I would do normally on any site i'm building

**Comment by retnonindya** [comment]
+1 ✨  This is also a feature request. It will be great to have the setting "Open on click" and "Show arrow" consistent with the menu, overlay or not. If the setting is not being applied on Overlay menu, perhaps we can hide/remove the toggle buttons or put an information/tooltip on it? Something like: "The settings below will only apply when Overlay Menu is off" 🤔

**Comment by masperber** [comment]
+1 on this feature request. It would be great to have the option to have the option to collapse and expand submenus within the overlay menu.

**Comment by csabarakasz** [comment]
+1 on this feature request in the name of one of our users. They would like submenus to be possible to collapse and uncollapse in mobile devices as well. Thanks.

**Comment by Jaccuse** [comment]
+1 on this request after speaking with a couple of users who would like to see mobile submenus have a collapsible option, including myself!

**Comment by gaswirth** [comment]
More than 6 months later, I'm still running into this issue all the time. Just had to hack around it again today. If anyone is interested, I'm happy to share my very simple custom JS I've been reusing to manually add this feature, but if you're here talking about it, my guess it's likely you're already doing something similar 😋

**Comment by carolinan** [comment]
The best way to share your solution is to create a pull request for the issue in this repository.

**Comment by gaswirth** [comment]
> The best way to share your solution is to create a pull request for the issue in this repository.

Agreed, if my solution weren't just a frontend hack. It doesn't address the block editor -- just a simple bit of JS to allow submenus to collapse when clicked. It's definitely not a real solution.

**Comment by tanjoymor** [comment]
+1 submenus being auto-collapsed in the overlay menu is a high request of users. Provided the Navigation block is **not** using the “Page List” block the following CSS does fix it in Block themes:

```
/* Make mobile menu display sub-menu on hover only  */
@media only screen and (max-width: 599px) {
    .wp-block-navigation.is-responsive li ul.wp-block-navigation__submenu-container  {
        display: none;
    }
    .wp-block-navigation.is-responsive li:hover > ul.wp-block-navigation__submenu-container {
        display: block;
    }
}
```

But making it a user controlled setting would be ideal. We could introduce it as a setting that needs to be activated, so that it doesn’t automatically impact existing users. But I also think that there is a higher use case of wanting the subitems to auto-collapse and therefore turning it on by default is likely fine, all we need to include is a pop-up notification that informs users of the change and provide a direct jump link to the setting where they can turn it off if desired. That transparency and awareness notification will satisfy most users so that it’s not an unexpected or jarring change. 

But this would be a huge improvement in the area of being intuitive. Auto-collapsed mobile menus is far more the norm than the current setup.

**Comment by carolinan** [comment]
Please consider that hover is a mouse only feature, it does not work on mobile devices or touch screens, so if anyone plans to use this code, you need to be aware that it is not a complete solution.

**Comment by tanjoymor** [comment]
> Please consider that hover is a mouse only feature, it does not work on mobile devices or touch screens, so if anyone plans to use this code, you need to be aware that it is not a complete solution.

Is there an aspect that is perhaps browser or device dependent? Because this did work on Chrome for iPhone the last time I tested it.

**Comment by pareshsojitra** [comment]
I just fixed this doing some tweak. if anyone looking for this. without blocking main menu link.

I can also share if anyone still looking to this.

I have done this in latest WordPress Theme.

![image](https://github.com/WordPress/gutenberg/assets/5764334/3def3de4-073e-4a49-aeb3-12f21d6518f0)

CSS:
`
@media (max-width:599px) {
	header .wp-block-navigation .wp-block-navigation-item {
		justify-content: center;
	}
	header .wp-block-navigation ul li {
		width: 100%;
	}
	header .wp-block-navigation .wp-block-navigation__submenu-icon {
		height: auto;
		width: auto;
		position: absolute;
		border: 1px solid #ccc;
		padding: 3px;
		right: 0;
		top: 4px;
	}
	header .wp-block-navigation__responsive-container.is-menu-open .wp-block-navigation__responsive-container-content .has-child .wp-block-navigation__submenu-container {
		display: none;
		width: 100%;
	}
	header .wp-block-navigation__responsive-container.is-menu-open .wp-block-navigation__responsive-container-content .wp-block-navigation__submenu-icon {
		display: block;
	}
}
`

jQuery:
`
/* + Document On Ready */
$(document).on("ready", function() {
	$(".wp-block-navigation-submenu__toggle").on("click", function() {
		var li = $(this).parent();
		if ( li.hasClass("ddl-active") || li.find(".ddl-active").length !== 0 || li.find(".wp-block-navigation-submenu").is(":visible") ) {
			li.removeClass("ddl-active");
			li.children().find(".ddl-active").removeClass("ddl-active");
			li.children(".wp-block-navigation-submenu").slideUp();
		}
		else {
			li.addClass("ddl-active");
			li.children(".wp-block-navigation-submenu").slideDown();
		}
	});
});
`

**Comment by pareshsojitra** [comment]
Here is how it works without any issue. I hop this considered to be added in WordPress Core ? Anyone can do this how.

https://github.com/WordPress/gutenberg/assets/5764334/70dd7688-dd1c-48da-bf6a-26857683690a

**Comment by carolinan** [comment]
I don't think using jQuery is the solution that is right for WordPress core.

**Comment by pareshsojitra** [comment]
Dear Carolina,

Thank you for your response and consideration. I understand the concern regarding the use of jQuery in the WordPress core. I appreciate your dedication to maintaining best practices and accessibility standards.

I would like to further discuss the issue and explore alternative solutions that align more closely with WordPress conventions. If you have any suggestions or specific guidelines that I can follow to address the submenu toggle problem without relying on jQuery, I'm eager to learn and adapt the solution accordingly.

I believe in the collaborative spirit of the WordPress community, and I'm open to feedback and constructive criticism. Let's work together to find the best approach that not only resolves the problem but also adheres to the established standards.

Looking forward to your guidance and input.

**Comment by pareshsojitra** [comment]
Dear Carolina,

In addition to the submenu toggle issue, I'm also working on improving the mobile menu experience. Specifically, I'm implementing a solution where, in mobile screens, menu items that exceed the screen width will be moved to a "More" menu for better usability.

I've successfully implemented these enhancements in my Bootstrap-based WordPress themes over the past eight years. Now, with the transition to Block Themes, I'm committed to adapting and contributing to ensure a seamless and user-friendly experience.

If you have any specific guidelines or recommendations regarding mobile menu improvements in Block Themes, I'd love to hear them. Your expertise and guidance are invaluable as I navigate this transition.

Thank you for your time and consideration.

https://github.com/WordPress/gutenberg/assets/5764334/f22d9bda-dc46-4748-b833-3ceb5ad04b79

**Comment by carolinan** [comment]
I am not trying to be dismissive.
The only reason why this is not looked at more closely and actively is that many full time contributors are working on other things, changes that are already planned to be part of a WordPress release. Often, things will not move as fast as we wish.

There was a major change to the overlay menu that was planned for WordPress 6.5, that was not completed in time, but that will hopefully be part of WordPress later this year. It does not specifically address collapsing sub menus, it does affect how the overlay menu works: https://github.com/WordPress/gutenberg/issues/43852#issuecomment-1884949063

**Comment by pareshsojitra** [comment]
Thank you for taking the time to provide further context. I completely understand the challenges of managing contributions and the priorities that full-time contributors have.

I appreciate your insights on the upcoming changes planned for WordPress 6.5, particularly the major change to the overlay menu. While it may not directly address collapsing submenus, I look forward to seeing how it affects the overall user experience.

I understand that the WordPress development process has its pace, and I appreciate the hard work that the team puts into planned releases. If there's any way I can contribute or assist with ongoing efforts, please let me know. I'm eager to collaborate and help ensure a smooth transition for the community.

Thank you for your dedication and for keeping me informed about the ongoing developments.

**Comment by neverthejetsetter** [comment]
This is probably not a solution that would be integrated into Wordpress Core, but it kinda works for me.
I took the jQuery and CSS that @pareshsojitra shared and re-worked it via Chat GPT to develop a Javascript (and CSS solution):

javascript:
```
document.addEventListener('DOMContentLoaded', function () {
  // Check screen width on load and resize
  function checkScreenWidth() {
    const screenWidth =
      window.innerWidth ||
      document.documentElement.clientWidth ||
      document.body.clientWidth;
    return screenWidth < 600;
  }

  function initMenuListeners() {
    const submenuToggles = document.querySelectorAll(
      '.wp-block-navigation-submenu__toggle'
    );

    submenuToggles.forEach((toggle) => {
      toggle.addEventListener('click', function (event) {
        event.preventDefault(); // Prevent the default action of the link

        const listItem = toggle.closest('li'); // Get closest parent list item (li)
        const submenu = listItem.querySelector('.wp-block-navigation-submenu');

        const isSubMenuOpen = listItem.classList.toggle('ddl-active');

        // Close other open submenus at the same level
        closeSiblingSubmenus(listItem);

        // Toggle display of the submenu based on 'ddl-active' class
        if (submenu) {
          submenu.style.display = isSubMenuOpen ? 'block' : 'none';
        }
      });
    });

    // Function to close sibling submenus
    function closeSiblingSubmenus(currentListItem) {
      const parentUl = currentListItem.parentElement; // Get the parent UL
      const siblingItems = parentUl.querySelectorAll('li.has-child');

      siblingItems.forEach((item) => {
        if (item !== currentListItem) {
          item.classList.remove('ddl-active');
          const submenu = item.querySelector('.wp-block-navigation-submenu');
          if (submenu) {
            submenu.style.display = 'none';
          }
        }
      });
    }
  }

  // Initial check and attach listeners
  if (checkScreenWidth()) {
    initMenuListeners();
  }

  // Re-attach listeners on window resize
  window.addEventListener('resize', function () {
    if (checkScreenWidth()) {
      initMenuListeners();
    }
  });
});
```


CSS (mostly stayed the same):
```
@media (max-width: 599px) {
  header .wp-block-navigation .wp-block-navigation-item {
    justify-content: center;
  }
  header .wp-block-navigation ul li {
    width: 100%;
  }
  header .wp-block-navigation .wp-block-navigation__submenu-icon {
    height: 36px;
    width: auto;
    position: absolute;
    border: 0px solid #ccc;
    padding: 5px;
    left: 60vw;
    top: 0px;
  }
  header
    .wp-block-navigation__responsive-container.is-menu-open
    .wp-block-navigation__responsive-container-content
    .has-child
    .wp-block-navigation__submenu-container {
    display: none;
    width: 100%;
  }
  header
    .wp-block-navigation__responsive-container.is-menu-open
    .wp-block-navigation__responsive-container-content
    .wp-block-navigation__submenu-icon {
    display: block;
  }
}
```

![image](https://github.com/user-attachments/assets/aaa3cbe8-1eb6-458f-853d-ce3c14b175ac)

![image](https://github.com/user-attachments/assets/5a5b1660-c760-44aa-ab53-cb3dfad5ca79)


These snippets will allow the submenu/child element to open when a parent link (or icon) is clicked. And it will automatically close the current child element when another parent element is clicked. It's not perfect, but it gets the job done for me.

For the Navigation settings, I left most settings default, so you're experience my vary. But I've included a screenshot so you can see what I'm using.

I'm running this on the latest Wordpress version (v6.6.2) and using the most recent Twenty-Twenty-Four theme (v1.2).

I simply added the two snippets  via the WPCode Lite plugin.

Thanks for the help @pareshsojitra.

I'm really looking forward to the new Wordpress overlay menu as well and hoping that it addresses the expanded navigation submenus in a future release.

**Comment by annezazu** [comment]
> As a user, the one thing, I think that is holding back the adoption of block themes is the navigation block, which is why I have been regularly checking for what’s coming in 6.7.

>Mobile menus is the only area we have had negative feedback on our transition form a classic to a block theme site.

>In the current FSE Navigation Block implementation – submenus on mobile do not collapse, which leads to very very long and ungainly menus. Here is my current menu which is long and unwieldy. See:
https://www.decimus.com/wp-content/uploads/2024/10/Mobile_Menu_Block_Ollie.jpg

Pulling this feedback in[ from the Roadmap to 6.7 post. ](https://make.wordpress.org/core/2024/09/03/roadmap-to-6-7/#comments)

**Comment by hagege** [comment]
I also think that it is time to take a closer look at this issue. The answers in this issue and also discussions with customers show that long submenus are not uncommon and that an option should be added that allows submenus to be expanded or not.

**Comment by carolinan** [comment]
@hagege One of the reasons why this is not worked on right now is that the **intention** is to overhaul how the overlay works and use a template part for the overlay, one that can have more editing options.
Because this change is already planned, and work has already started a few releases ago, it is reasonable to postpone the suggestion in this issue until after this change.

**Comment by hagege** [comment]
Thank you for your answer.
In this context, I would like to point out a new option in the “Options for Block Themes” plugin (https://wordpress.org/plugins/template-editor/#developers). The developer has implemented my feature request for this plugin in the current version.  
I have described this new option in the following post: https://haurand.com/shrinked-header-bei-twenty-twenty-four/#:~:text=Aufklappbare%20Untermen%C3%BCs%20bei%20Block%20Themes (unfortunately only in German).

**Comment by DecimusBurton** [comment]
@hagege thanks for posting the link to Options for Block Themes plugin which we have implemented on our hobby site [BajanThings](https://www.bajanthings.com/category/barbados-ww1-to-ww2/).  It solves the problem of [long and unwieldy menus on mobile devices](https://www.decimus.com/wp-content/uploads/2024/10/Mobile_Menu_Block_Ollie.jpg).  Using the Options for Block Themes plugin on BajanThings with mobile menus, the only thing that fails to work is the CSS magic we implemented that gives a visual cue if a submenu is active and would also highlight the sub-menu parent menu (you can see this in action on the desktop link I posted if you hover on the parent) - I guess we cannot have everything!

```
/* Ollie Navigation - Highlight parent page link when on child page */
.wp-block-navigation-item:has(.current-menu-item)  {
   color: #4D34FA !important;
}
```
Well we can have everything! Fixed with some additional global CSS

```
/*This highlights the parent page link when on a child page when using the: Options for Block Themes plugin by Oliver Campion of Web:D which has an option for expandable submenus to the Navigation Block modal */
.wp-block-navigation.has-expandable-modal-submenus .wp-block-navigation-item:has(.current-menu-item)>a  {
   color: #4D34FA;
}
```

See it in action here where there is a [photo](https://www.bajanthings.com/wordpress-6-7-first-impressions/#webd).

**Comment by hagege** [comment]
@DecimusBurton 
Thank you for the information and the reference to my post about the plugin. There are indeed more possibilities with this plugin. Unfortunately, the design options are still quite limited compared to many classic themes. I very much hope that there will be some improvements in the future. In my opinion, navigation is one of the biggest issue when using block themes.

**Comment by DecimusBurton** [comment]
Hans I agree re navigation and block themes.  It seems we are the minority.

**Comment by frspp** [comment]
+1

IMO no way mobile menu can have all items open by default, its easily sooooooo long, this image shows well: [long and unwieldy menus on mobile devices](https://www.decimus.com/wp-content/uploads/2024/10/Mobile_Menu_Block_Ollie.jpg).

**Comment by lemonroti** [comment]
+1 , this is a needed featured for all website especially woocommerce site ! Such an important featured should be included in wp site

**Comment by adithya-naik** [comment]
#76890 solves this issue

**Comment by carolinan** [comment]
@adithya-naik  @t-hamano Looking closer at this, I think this is already solved?
Maybe @hagege you can fill me in what I am missing?

**Outside** the navigation overlay, when you add a submenu to the navigation block, the navigation block has the following options:
Submenus >Submenu Visibility: Hover or Click, and then "show arrow":
<img width="267" height="204" alt="Image" src="https://github.com/user-attachments/assets/bed2bfdb-1ca5-400c-b1b7-886b05412c11" />

But the navigation block that is **inside** the navigation overlay **does not automatically inherit this setting.**

The navigation overlay is a template part and the navigation block inside it, is a separate block.
It **defaults** to a third setting: Always.
Submenus >Submenu Visibility: Always. 
I assume this is to be compatible with the old version of the navigation block.

So to make the submenu items inside the navigation overlay  hidden by default and revealed on hover or click, the
user needs to select one of those options:

<img width="284" height="186" alt="Image" src="https://github.com/user-attachments/assets/c220e4b4-c7b2-417f-a7f8-3774a3f4c40e" />

**Comment by hagege** [comment]
@carolinan 
Thanks for looking at this.
Sorry if I haven’t explained this clearly enough: 
The point is that there are scenarios where a single menu item has many sub-menu items. In the desktop view, these sub-menu items can be expanded either by hovering over them or by clicking on them. 
In the mobile view, however, all submenu items are always expanded when a submenu is selected. If there are a lot of submenu items, you may have to scroll down quite a long way on mobile to see further menu items. This can be annoying.
The Ollie block theme already includes an option to ensure that submenus are not automatically expanded, but collapsed.
I believe an option of this kind would also be desirable for the standard navigation block.

![Image](https://github.com/user-attachments/assets/ff1be83c-f2de-4ac8-95da-08b945ffab93)
![Image](https://github.com/user-attachments/assets/968d5c83-dcd1-472f-94e9-43c870001241)

As it stands, I find this confusing because it works with overlays but not with the standard navigation block, even though it isn’t explicitly clear that this option is currently unavailable for the standard navigation block. 
But I’m looking forward to the solution from @adithya-naik

**Comment by adithya-naik** [comment]
@hagege Sure, iam working on this

**Comment by hagege** [comment]
@adithya-naik 
thanks for looking at this issue. 👍🏻

**Comment by carolinan** [comment]
@hagege Looking at the screenshot, it is showing the old controls, not the controls for the navigation block that are in Gutenberg and will be in WordPress 7.0.

Again, changing the default without the user changing any settings, is a breaking change for existing websites.

**Comment by hagege** [comment]
@carolinan 
That’s right. I was referring to the current situation in WordPress 6.9.4. For clarity, here’s a screenshot in WordPress 7: 

![Image](https://github.com/user-attachments/assets/8f13a870-78fe-4cdc-aad3-34d759bc6831)

**Comment by t-hamano** [comment]
I agree that the current behavior, while intentional, is indeed confusing and unclear. We must maintain full backward compatibility, but there might be room for improvement in the UI and functionality.

These are fragmented ideas. Please note that these suggestions are based on the specifications of the unreleased WordPress 7.0 RC or the Gutenberg plugin.

### When the Overlay Visibility setting is "Off"

There should be no issues. The submenu visibility will always function correctly.

<img width="292" height="382" alt="Image" src="https://github.com/user-attachments/assets/758ce498-e964-4b95-91a2-ad9e835a25b0" />

### When the Overlay Visibility setting is "Mobile" and the Overlay Template is "Default"

We might add a setting for submenus in the mobile menu. However, to maintain backward compatibility, the default value must be "Always".

| Before | After |
|--------|--------|
| <img width="276" height="387" alt="Image" src="https://github.com/user-attachments/assets/1c691a68-ca09-4916-8e2a-2133d72a1534" /> | <img width="280" height="498" alt="Image" src="https://github.com/user-attachments/assets/1c2e088e-5ab6-4e59-b239-f7f6b8b24145" /> |

### When the Overlay Visibility setting is "Always" and the Overlay Template is "Default"

I believe the current issue is that the submenu within the overlay always expands, even though the 'Submenu Visibility' setting is either "Hover" or "Click". While this might be fixable, if a user _accidentally_ sets it to "Hover" or "Click", The menu behavior within the overlay is unexpectedly changing.

<img width="287" height="411" alt="Image" src="https://github.com/user-attachments/assets/45fb2fda-97be-4d99-b131-8a362aed76e0" />

### When the Overlay Visibility setting is "Mobile" and the Custom Navigation Overlay Template is used

We may need to explain that the Submenu Visibility setting does not actually affect the navigation overlay.

| Before | After |
|--------|--------|
| <img width="278" height="392" alt="Image" src="https://github.com/user-attachments/assets/49ad92c1-ddfc-4972-aae0-adaa4095b61a" /> | <img width="272" height="433" alt="Image" src="https://github.com/user-attachments/assets/0e67fdbe-9e9e-4b15-bf33-4ee40ae69734" /> |

### When the Overlay Visibility setting is "Always" and the Custom Navigation Overlay Template is used

The submenu behavior is determined by the navigation block inserted within the navigation overlay template, so the Submenu Visibility setting should not be necessary here.

| Before | After |
|--------|--------|
| <img width="279" height="449" alt="Image" src="https://github.com/user-attachments/assets/a7de38a1-7817-48cd-a355-37f4f195aad0" /> | <img width="272" height="189" alt="Image" src="https://github.com/user-attachments/assets/13dee89b-4c2a-4122-8cba-01edc437cc45" /> |

**Comment by adithya-naik** [comment]
I understand the backward compatibility concern. I'd like to work on Scenario 2/3. Should I wait for the design discussion to conclude before opening a new PR?

**Comment by t-hamano** [comment]
> Should I wait for the design discussion to conclude before opening a new PR?

I think so. This is because it is still under discussion and a clear approach has not yet been decided. Design feedback will also be necessary.

**Comment by adithya-naik** [comment]
> > Should I wait for the design discussion to conclude before opening a new PR?
> 
> I think so. This is because it is still under discussion and a clear approach has not yet been decided. Design feedback will also be necessary.

Sure @t-hamano, Thanx meanwhile please look into my other PRs, that would be greatful 

And assign this to me

**Comment by le-sgs** [comment]
Hi @t-hamano or @adithya-naik, I was wondering if you folks knew of any further updates on this issue?

**Comment by t-hamano** [comment]
@le-sgs Let's wait for the design feedback. I don't think we need to rush the resolution.

**Comment by frspp** [comment]
duplicate issue https://github.com/WordPress/gutenberg/issues/81543 and PR https://github.com/WordPress/gutenberg/pull/81585

Hope PR gets accepted as current behauviour on mobile overlay (submenus always open) is very unexpected and confusing for admin too, imo.

**Comment by t-hamano** [comment]
> duplicate issue https://github.com/WordPress/gutenberg/issues/81543 and PR https://github.com/WordPress/gutenberg/pull/81585

As there are many discussions ongoing on this issue, let's close #81543 as a duplicate.

**Comment by frspp** [comment]
Is there a way to help this move this forward? Where is the design feedback taking place?

**Comment by t-hamano** [comment]
In my opinion, I would advise against changing the submenu default behavior inside the overlay, as it breaks backward compatibility. We should not force a change in this behavior without the user's explicit opt-in. Furthermore, users can now have full control over the navigation behavior within the overlay by using a custom overlay template part.

Instead, what we might want to solve here is adding some clarifying notes to the Navigation block sidebar or hiding unnecessary UI depending on the context.

**Comment by frspp** [comment]
https://github.com/WordPress/gutenberg/issues/44346#issuecomment-4168578681

# 1. Show hint always when editing main navigation 
To keep things simple and understandable, I suggest showing the suggested Submenu Visibility hint text "Does not affect navigation within the navigation overlay." *always* except when editing the Navigation Overlay. 

## 1b. To reduce confusion, add hint also when editing the Navigation Overlay
When editing Navigation Overlay, most likely not everybody knows what they are doing. So again give hint under Submenu Visibility: "Affects only navigation in the navigation overlay."

# 2. When editing the Navigation Overlay: Remove option Hover
When editing the Navigation Overlay: AFAIK on mobile there is no such thing as hover, so remove option "Hover".

After:
<img width="958" height="462" alt="Image" src="https://github.com/user-attachments/assets/bb7c5b4e-d9ef-49bf-91f7-7eb8d763492a" />

# 3. BUG: Navigation Overlay: Submenu Visibility > Click does not work
On live site submenus/child pages are always shown, even if the Submenu Visibility is set to Click. Works only on the editor preview window, but not on live site. 

Preview: Seems to work
<img width="952" height="545" alt="Image" src="https://github.com/user-attachments/assets/3fd42488-579a-4f6a-82bd-06afd17bf324" />

Live site: Broken
<img width="600" height="448" alt="Image" src="https://github.com/user-attachments/assets/be7e0d6d-e01d-4985-94e5-60c914f1257a" />

Also noted an overlap bug in that last screenshot https://github.com/WordPress/gutenberg/issues/83319
